openhtf/util/units/bin/units_from_xls.py

```diff
@@ -49,20 +53,107 @@
                 'Common\nCode',
                 'Symbol']
 
-# The contents of the lines of the output file between which the generated code
-# will be inserted.
-START_DELIMITER = '######## BEGIN_UNITS ########'
-END_DELIMITER = '######## END_UNITS ########'
+PRE = '''# coding: iso-8859-1
+# THIS FILE IS AUTOMATICALLY GENERATED. DO NOT EDIT.
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Units of measure for OpenHTF.
+
+Used to retrieve UNECE unit codes by object, name, or suffix:
+
+    from openhtf.util import units
+
+    # The following three lines are equivalent:
+    @measures(units.METRE_PER_SECOND)
+    @measures(units.Unit('m/s'))
+    @measures(units.Unit('metre per second'))
+
+OpenHTF uses UNECE unit codes internally because they are relatively complete
+and modern, and because they are recognized internationally. For full details
+regarding where we get the codes from and which units are avaiable, see the
+docstring at the top of openhtf/util/units/bin/units_from_xls.py.
+"""
+
+
+import collections
+
+
+UnitInfo = collections.namedtuple('Unit', 'name code suffix')
+
+ALL_UNITS = []
+
+
+# NO_DIMENSION means that there are units set, but they cannot be expressed
+# by a known dimension (such as a ratio)
+NO_DIMENSION = UnitInfo('No dimension', 'NDL', None)
+ALL_UNITS.append(NO_DIMENSION)
+
+NONE = UnitInfo('None', None, None)
+ALL_UNITS.append(NONE)
+'''
+
+POST = '''
+# Convenience aliases.
+MINUTE = MINUTE_UNIT_OF_TIME
+ALL_UNITS.append(MINUTE)
```

Comment:
This doesn't do anything since it's already in `ALL_UNITS` and everything else uses it in a dict, no? I'd `del ALL_UNITS` at the end since it doesn't make sense to use it directly, too.